app/src/main/java/org/mozilla/fenix/library/history/HistoryViewItemFlow.kt

```diff
@@ -0,0 +1,153 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.library.history
+
+import android.content.res.Resources
+import androidx.paging.Pager
+import androidx.paging.PagingConfig
+import androidx.paging.PagingData
+import androidx.paging.cachedIn
+import androidx.paging.filter
+import androidx.paging.insertFooterItem
+import androidx.paging.insertSeparators
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.map
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.service.fxa.manager.FxaAccountManager
+import org.mozilla.fenix.R
+import org.mozilla.fenix.components.history.PagedHistoryProvider
+
+/**
+ * Flow of [HistoryViewItem] items used in [HistoryFragment] screen to populate the history list
+ * in [HistoryAdapter].
+ *
+ * @param historyProvider Data source of history and history group items.
+ * @param browserStore Provides information about numbers of recently closed tabs.
+ * @param isRemote The dataSource can provide local, remote and mixed history.
```

Comment:
if it's `true` than the history would be from other devices, and if it's `null` it would be both. Do you think using an `enum` for it would be better?

Reply:
You'll need to document this everywhere that uses the same `isRemote`.  But yes, I would say a enum is probably more readable.